addons/common/functions/fnc_adjustMagazineAmmo.sqf

```diff
@@ -0,0 +1,69 @@
+#include "..\script_component.hpp"
+/*
+ * Author: Katalam, Blue, Brett Mayson, johnb43
+ * Handle adjusting a magazine's ammo
+ *
+ * Arguments:
+ * 0: Vehicle or Unit <OBJECT>
+ * 1: Item <STRING>
+ * 2: Ammo to add <NUMBER> (default: -1)
+ *
+ * Return Value:
+ * Empty mag <BOOLEAN> - true if the magazine is empty after the ammo is consumed
+ *
+ * Example:
+ * [player, "30Rnd_556x45_Stanag", 1] call ace_common_fnc_adjustMagazineAmmo;
+ *
+ * Public: No
+ */
+
+params ["_unit", "_magazine", ["_count", -1]];
+
+private _containers = if (_unit isKindOf "CAManBase") then {
+    [uniformContainer _unit, vestContainer _unit, backpackContainer _unit]
+} else {
+    [_unit]
+};
+
+scopeName "main";
+
+private _container = objNull;
+private _magazinesContainer = [];
+private _newAmmoCount = 0;
+
+{
+    _container = _x;
+
+    // Get all magazines of _magazine type
+    _magazinesContainer = (magazinesAmmoCargo _container) select {_x select 0 == _magazine};
+
+    // If there are none, skip to next container
+    if (_magazinesContainer isEqualTo []) then {
+        continue;
+    };
+
+    // Get the ammo count
+    _magazinesContainer = _magazinesContainer apply {_x select 1};
+
+    // Sort, smallest first
+    _magazinesContainer sort true;
+
+    {
+        _newAmmoCount = _x + _count;
+
+        if (_newAmmoCount >= 0) exitWith {
```

Comment:
Come to think of this, this doesn't support adding ammo properly: It doesn't check if the magazine is overfilled - which isn't even possible in game, so if you call this function with params to add ammo when all magazines are full, the outcome is that nothing will be added.
When rewriting the function, I focused on ammo removal, I completely ignored the adding part.

By the looks of it, the old function didn't support it either. I didn't test it in detail, but looking at the code suggests it wasn't possible.

The question is: Do we want to support both adding and removing ammo? Do we want to make it, so that it adds/removes ammo over multiple not-full magazines? Do we want to add extra magazines when adding ammo and all existing magazines are full?

Reply:
I think it should try to add extra mags yes.

Should also test that it correctly handles the case where we need to take from multiple magazines. If I have two mags with 5 ammo each, doing -7 should leave me with a single mag with 3 ammo remaining

Reply:
I think returning if the command managed to add/remove the requested ammo successfully might be more useful. We could return both though, as an array.